File: src/utils/conv/rss3writer.c
#include <libr3r.h>
#include <stdio.h>

#include "r3r-conv.h"
#include "rss3writer.h"

void write_rss3(void * item)
{
	FILE * fh = get_handle();
	if (!fh) return;

	fprintf(fh, "title: %s\nlink: %s\ndescription: %s\nSubject: %s\nGenerator: %s\nLanguage: %s\nGUID: %s\nRights: %s\n\n",
			ri(item, "title"), ri(item, "link"), ri(item, "description"), ri(item, "subject"),
			ri(item, "generator"), ri(item, "language"), ri(item, "id"), ri(item, "copyright"));

	if (out_file)
		fclose(fh);
}

File: src/utils/conv/rss3writer.pas
unit Rss3Writer;

interface

uses
  Conv, FeedItem;

procedure WriteRss3(Item: TFeedItem);

implementation

procedure WriteRss3(Item: TFeedItem);
var
  fh: PText;
begin
  fh := GetFileHandle;
  WriteLn(fh^, 'title: ', Item.Title, LineEnding, 'link: ', Item.Link,
    LineEnding, 'description: ', Item.Description, LineEnding, 'subject: ',
    Item.Subject, LineEnding, 'generator: ', Item.Generator, LineEnding,
    'language: ', Item.Language, LineEnding, 'guid: ', Item.Id, LineEnding,
    'rights: ', Item.Copyright, LineEnding, LineEnding);

  if OutFile <> '' then
  begin
    Close(fh^);
  end;

  Dispose(fh);
end;

end.
